snuba/query/query.py

```diff
@@ -273,6 +273,9 @@ def get_limitby(self) -> Optional[Limitby]:
     def get_sample(self) -> Optional[float]:
         return self.__body.get("sample")
 
+    def set_sample(self, rate: Optional[float]) -> None:
+        self.__body["sample"] = rate
```

Comment:
All this will be in the ClickhouseAST

Reply:
I don't understand this comment. This query and the AST query don't share a common base and aren't substitutable.